ls1intum/Artemis
# [Bugfix] Sentry Issue Type Error in programming submission service

### Checklist

### Motivation and Context


This PR solves the [Sentry Issue ARTMIS-26](https://sentry.ase.in.tum.de/organizations/artemis/issues/71/?project=2)

### Description

Error: TypeError call(src/main/webapp/app/exercises/programming/participate/programming-submission.service)
Cannot read property 'next' of undefined.
I inspected the use of `.next()` at other places in the respective service and adapted two occurrences of `.next()` without prior null check.

### Steps for Testing


As I could not reproduce the issue, I would say following procedure is enough to test the changes:

-> automatic assessment

1. Log in to Artemis
2. Search a programming exercise without due date and manual correction enabled
3. As a student, participate in that programming exercise  and submit a solution 
4. inspect console log, are there any TypeErrors? Is the automatically created result as expected?
5. As a tutor/instructor, assess the submission and save the assessment

-> manual assessment

1. Log in to Artemis
2. Search a programming exercise with due date and manual correction enabled
3. As a student, participate in that programming exercise and submit a solution 
4. Change the due date to a date in the past (or wait if the due date is within the next minutes)
5. inspect console log, are there any TypeErrors? Is the automatically created result as expected?
6. As a tutor/instructor, assess the submission and save the assessment
